contracts/Overtime/FreeBetsHolder.sol

```diff
@@ -0,0 +1,164 @@
+        uint _expectedPayout,
+        uint _additionalSlippage,
+        address _referrer,
+        address _collateral
+    ) external notPaused canTrade(msg.sender, _collateral, _buyInAmount) {
+        balancePerUserAndCollateral[msg.sender][_collateral] -= _buyInAmount;
+        address createdTicket = sportsAMM.trade(
+            _tradeData,
+            _buyInAmount,
+            _expectedPayout,
+            _additionalSlippage,
+            address(0),
+            _referrer,
+            _collateral == address(sportsAMM.defaultCollateral()) ? address(0) : _collateral,
+            false
+        );
+        ticketToUser[createdTicket] = msg.sender;
+    }
+
+    function tradeLive(
+        bytes32 _gameId,
+        uint16 _sportId,
+        uint16 _typeId,
+        uint8 _position,
+        uint _buyInAmount,
+        uint _expectedPayout,
+        uint _additionalSlippage,
+        address _referrer,
+        address _collateral
+    ) external notPaused canTrade(msg.sender, _collateral, _buyInAmount) {
+        bytes32 _requestId = liveTradingProcessor.requestLiveTrade(
+            _gameId,
+            _sportId,
+            _typeId,
+            _position,
+            _buyInAmount,
+            _expectedPayout,
+            _additionalSlippage,
+            address(0),
+            _referrer,
+            _collateral
+        );
+        liveRequestsPerUser[_requestId] = msg.sender;
+    }
+
+    function confirmLiveTrade(
+        bytes32 requestId,
+        address _createdTicket,
+        uint _buyInAmount,
+        address _collateral
+    ) external notPaused {
+        address _user = liveRequestsPerUser[requestId];
+        require(_user != address(0), "Unknown live ticket");
+
+        if (_collateral == address(0)) {
+            _collateral = address(sportsAMM.defaultCollateral());
+        }
+
+        require(supportedCollateral[_collateral], "Unsupported collateral");
+        require(balancePerUserAndCollateral[_user][_collateral] >= _buyInAmount, "Insufficient balance");
+
+        balancePerUserAndCollateral[_user][_collateral] -= _buyInAmount;
+        ticketToUser[_createdTicket] = _user;
+    }
+
+    function claimTicket(address _ticket) external notPaused {
+        address _user = ticketToUser[_ticket];
+        require(_user != address(0), "Unknown ticket");
+        IERC20 _collateral = Ticket(_ticket).collateral();
+        uint balanceBefore = _collateral.balanceOf(address(this));
+        sportsAMM.exerciseTicket(_ticket);
+        uint balanceAfter = _collateral.balanceOf(address(this));
+        uint exercized = balanceAfter - balanceBefore;
+        if (exercized > 0) {
+            //TODO: can it happen that user claims less than what he paid?
+            uint earned = exercized - Ticket(_ticket).buyInAmount();
+            IERC20(_collateral).safeTransfer(_user, earned);
+            balancePerUserAndCollateral[_user][address(_collateral)] += Ticket(_ticket).buyInAmount();
+        }
```

Comment:
shouldn't happen even if all games are cancelled, since in V2 we extract that after.